sdk/storage/Azure.Storage.Common/src/StorageExceptionExtensions.cs

```diff
@@ -107,5 +127,25 @@ private static string GetErrorCode(Response response, string errorCode)
 
             return messageBuilder.ToString();
         }
+
+        /// <summary>
+        /// Check if a Response will throw an exception if you try to access
+        /// its Value property.
+        /// </summary>
+        /// <typeparam name="T">Type of the Response Value.</typeparam>
+        /// <param name="response">The response to check.</param>
+        /// <returns>True if the response will throw.</returns>
+        public static bool IsExplodingResponse<T>(this Response<T> response) =>
```

Comment:
I would call it IsNoBodyResponse && AsNoBodyResponse while exploding sounds fun I think "NoBody" describes what's happening better.

Reply:
I wanted a name that worked for this and batching's delayed responses.  That said, yeah, this might not survive a trip through policheck.  I'll rename it to something more general.

Reply:
I've changed this to `IsUnavailable` and `AsNoBodyResponse`.